### PR TITLE
Feature/156 add scope to run config

### PR DESCRIPTION
**Pull request by brianneoberson**
- created a "Default" run config which will be used for the actionable pitest runs 
- fixed the issue with the editing of run config settings. The editor now notices changes and they can be applied.

### CODE REVIEW

File: pitmutationmate/src/main/kotlin/com/amos/pitmutationmate/pitmutationmate/configuration/RunConfiguration.kt [hunk 1]
```diff
@@ -55,6 +57,13 @@ class RunConfiguration(
             logger.debug("MutationMateRunConfiguration: classFQN was updated to '$classFQN'.")
         }
 
+    override fun writeExternal(element: Element) {
+        super.writeExternal(element)
+        JDOMExternalizerUtil.writeField(element, "TASK_NAME", taskName)
+        JDOMExternalizerUtil.writeField(element, "GRADLE_EXECUTABLE", gradleExecutable)
+        JDOMExternalizerUtil.writeField(element, "CLASS_FQN", classFQN)
+    }
+
```
Comment (lheimbs) [inline pitmutationmate/src/main/kotlin/com/amos/pitmutationmate/pitmutationmate/configuration/RunConfiguration.kt:66]: Works but this seems like a weird work around.
What did work for me is setting the propertyName of the StoredProperty delegate in the RunConfigurationOptions correctly like so:

```
private var taskNameOption: StoredProperty<String?> = string("").provideDelegate(this, "taskName")
```

**Review by lheimbs — CHANGES_REQUESTED** [review]
I have a slight concern that users will add their custom class paths in the `Default` config then then wonder why they are gone after using some action...

**Review by nikomall34 — CHANGES_REQUESTED** [review]
I noticed that when I restart my IDEA, my run configuration persists, but the options I entered in the Gradle task and target classes are lost.

File: pitmutationmate/src/main/kotlin/com/amos/pitmutationmate/pitmutationmate/configuration/RunConfiguration.kt (hunk 1 above)
Comment (lheimbs) [reply pitmutationmate/src/main/kotlin/com/amos/pitmutationmate/pitmutationmate/configuration/RunConfiguration.kt:66]: I have added the respective names here: b07053b9d3f9f8e1a0a6b3f43a0ac36d3bbbba4f

**Comment by brianneoberson** [comment]
> I have a slight concern that users will add their custom class paths in the `Default` config then then wonder why they are gone after using some action...

I am afraid we won't have time to find and implement a solution for this until tomorrow. But this could be a small ticket for next sprint. We fixed the other requested changes.

**Review by nikomall34 — APPROVED** [review]

**Comment by nikomall34** [comment]
Maybe a quick fix could be to make the default runconfig not editable? Or maybe write somewhere in the Default run config "Please don't change".

**Review by lheimbs — APPROVED** [review]
I've added a simple solution to disable the `targetClasses` field when it is the default run config. All other configs' fields are still editable.

**Comment by nikomall34** [comment]
Unfortunately, I can still edit the default run configuration. I can simply modify the targetClasses in the run configuration, and when I click on "run" the specified classes are executed.

**Comment by lheimbs** [comment]
> Unfortunately, I can still edit the default run configuration. I can simply modify the targetClasses in the run configuration, and when I click on "run" the specified classes are executed.

Did you remove the old one? Because that one obviously doesn't have the attribute set to true

**Review by nikomall34 — APPROVED** [review]
Ok works as expected!

**Pull request merged**